core/src/main/java/org/infinispan/interceptors/impl/BaseStateTransferInterceptor.java

```diff
@@ -223,7 +225,7 @@ private Object handleReadCommandReturn(InvocationContext rCtx, VisitableCommand
       } else if (ce instanceof AllOwnersLostException) {
          if (trace)
             getLog().tracef("All owners for command %s have been lost.", cmd);
-         return null;
+         return rCommand.acceptVisitor(rCtx, LostDataVisitor.INSTANCE);
```

Comment:
That's a valid point. We should wrap the null into context, set skip lookup and add non-existent read version. This is a job for EWI, though, not STI.
And I wonder how to adapt scattered cache for that; In scattered mode the AllOwnersLostException usually leads to retrying with a new topology, because it is thrown any time the primary crashes but we haven't lost the data itself (we just don't know where it is). That's why we cannot wrap & fix null in EWI when we see AOLE.
An alternative would be doing a broadcast from ScatteredDistributionInterceptor, but that would partially duplicate the job of PrefetchInterceptor.
Any way, please log a JIRA and assign it to me.